packages/project-access/src/file/file-access.ts

```diff
@@ -66,3 +68,39 @@ export async function fileExists(path: string, memFs?: Editor): Promise<boolean>
         return false;
     }
 }
+
+/**
+ * Updates package.json file asynchronously by keeping the previous indentation.
+ *
+ * @param path - path to file
+ * @param packageJson - updated package.json file content
+ */
+export async function updatePackageJSON(path: string, packageJson: Package): Promise<void> {
+    await updateJSON(path, packageJson);
+}
+
+/**
+ * Updates manifest.json file asynchronously by keeping the previous indentation.
+ *
+ * @param path - path to file
+ * @param manifest - updated manifest.json file content
+ */
+export async function updateManifestJSON(path: string, manifest: Manifest): Promise<void> {
+    await updateJSON(path, manifest);
+}
+
+/**
+ * Updates JSON file asynchronously by keeping the indentation from previous content with new content for given path.
+ *
+ * @param path - path to file
+ * @param content - updated JSON file content
+ */
+async function updateJSON(path: string, content: object): Promise<void> {
+    // read old contents and indentation of the JSON file
+    const oldContentText = await readFile(path);
```

Comment:
If we add optional parameter `memFs?: Editor`, this line would then become 
```suggestion
    const oldContentText = await readFile(path, memFs);
```